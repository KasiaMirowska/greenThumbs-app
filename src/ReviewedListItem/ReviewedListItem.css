.small-img-container {
    /* max-width: 120px;
    max-height: 100px; */
    max-width: 40%;
    padding: 5px;
   
}

.home-list {
    display: flex;
    justify-content: flex-start;
    margin:auto;
    margin: auto;
    background-color: whitesmoke;
    margin-bottom: 30px;
    width: 95%;
    
}
.li {
    margin:auto;
    width: 95%;
}
.text {
    margin-left: 3%;

}
.thumb {
    margin-left: 30%;
    padding: 10px
}
@media(min-width: 800px) {
    
    .home-list {
        width: 85%;
        display: flex;
        margin:auto;
        margin-bottom: 30px;
     
    }
    .text {
        
        margin-left: 3%;
        padding: 20px;
    }
    .small-img-container {
        /* max-width: 120px;
        max-height: 100px; */
        max-width: 20%;
        padding: 20px;
       
    }
    .li {
        margin:auto;
        width: 80%;
    }
}