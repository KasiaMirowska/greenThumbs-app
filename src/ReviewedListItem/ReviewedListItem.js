import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faThumbsUp } from '@fortawesome/free-solid-svg-icons';
import TokenService from '../Services/token-service';
import config from '../config';

export default withRouter(function ReviewedListItem(props) {
   console.log(props)
    const greenThumbs = () => {
        const thumbs = [];
        for(let i =0; i < props.thumbs; i++){
            thumbs.push(<span key={i}>
                < FontAwesomeIcon icon={faThumbsUp} style={{ color: 'green', padding: '3px' }} />
            </span>)
        } 
        return(
            <div>
                {thumbs}
            </div>
        )
    }

    const openFullCard = (e) => {
        const token = TokenService.hasAuthToken(config.TOKEN_KEY)
        if(!token) {
            props.history.push('/login')
        } else {
            props.history.push(`/green_place/:${props.yelpId}/:${props.placeId}`)
        }

    }
    
    return (
        <div onClick={openFullCard}>
            <h3>{props.name}</h3>
            <p>{props.city}</p>
            {greenThumbs()}
        </div>
    )
});
