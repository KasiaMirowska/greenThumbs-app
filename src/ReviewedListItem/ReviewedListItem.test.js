import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as BR } from 'react-router-dom';
import ReviewedListItem from './ReviewedListItem';

describe('List component', () => {
    const props = {
        name: 'name',
        city: 'city',
        thumbs: [1,3,2],
        history: { push: '/' }
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDom.render(<BR><ReviewedListItem  {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const item = renderer.create(<BR><ReviewedListItem {...props} /></BR>);
        expect(item.toJSON()).toMatchSnapshot();
    })
})