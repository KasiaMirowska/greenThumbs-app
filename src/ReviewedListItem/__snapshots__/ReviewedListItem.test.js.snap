// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders UI as expected 1`] = `
<div
  className="home-list"
  onClick={[Function]}
>
  <div
    className="small-img-container"
  >
    <img
      alt="food presentation from the place"
    />
  </div>
  <div
    className="text"
  >
    <h3>
      name
    </h3>
    <br />
    <div
      className="text2"
    >
      <p>
        city
      </p>
      <p>
        Saved to category: 
      </p>
    </div>
    <div />
  </div>
</div>
`;
